FP_layout/app/src/main/java/com/example/dhyatmika/fp_layout/MapsActivity.java

```diff
@@ -36,11 +64,130 @@ protected void onCreate(Bundle savedInstanceState) {
      */
     @Override
     public void onMapReady(GoogleMap googleMap) {
+        this.fetchStations();
+
         mMap = googleMap;
 
         // Add a marker in Sydney and move the camera
-        LatLng sydney = new LatLng(-34, 151);
-        mMap.addMarker(new MarkerOptions().position(sydney).title("Marker in Sydney"));
-        mMap.moveCamera(CameraUpdateFactory.newLatLng(sydney));
+//        LatLng benhil = new LatLng(-6.217096, 106.815228);
+//        LatLng senayan = new LatLng(-6.227878, 106.800994);
+
+//        mMap.addMarker(new MarkerOptions().position(benhil).title("Bendungan Hilir"));
+//        mMap.addMarker(new MarkerOptions().position(senayan).title("Senayan"));
+
+//        CameraPosition cameraPosition = new CameraPosition.Builder()
+//                .target(senayan)
+//                .zoom(10).build();
+//
+//        mMap.animateCamera(CameraUpdateFactory.newCameraPosition(cameraPosition));
+
+        // set listener for marker click.
+        mMap.setOnMarkerClickListener(this);
+    }
+
+    @Override
+    public boolean onMarkerClick(final Marker marker) {
+
+        // retrieve the data from the marker
+        Integer clickCount = (Integer) marker.getTag();
+
+        Toast.makeText(this, marker.getPosition().toString(), Toast.LENGTH_SHORT).show();
```

Comment:
😸 Now comes the fun question, is `Helper.toastShort` and `Helper.toastLength` over-engineering When you can do one-liner like this? 

You are creating another function to replace this one-liner